File: microservices/first/utils/deployment_status.py
"""Utility module for interacting with the statuses of current first deployments"""
from enum import Enum, auto

class DeploymentStatus(Enum):
    """Represents the status of a deployment.

    STARTED - Deployment has been added to database but leader has not distributed work.
    IN_PROGRESS - Leader has start distributing work.
    DONE - Leader has ended its process.
    FAILED - Leader ran into an error
    """
    STARTED = auto()
    """The deployment has started but not yet built"""
    BUILDING = auto()
    """The deployment is currently building"""
    IN_PROGRESS = auto()
    """The deployment is currently running tests"""
    DONE = auto()
    """The deployment has completed successfully"""
    FAILED = auto()
    """The deployment has completed unsuccessfully"""

File: leaders/src/main/java/edu/cornell/status/DeploymentStatus.java
package edu.cornell.status;

/**
 * Represents the status of a deployment.
 */
public enum DeploymentStatus {
    /**
     * The leader has been added to the database but the leader has not yet distributed work.
     */
    STARTED,

    /**
     * The leader is building files.
     */
    BUILDING,

    /**
     * The leader has started the worker lifecycle testing process.
     */
    IN_PROGRESS,

    /**
     * The leader has completed.
     */
    DONE,

    /**
     * The leader has failed due to an error.
     */
    FAILED;

    /**
     * Returns just the enum constant name when converted to a string.
     *
     * @return The name of the enum constant.
     */
    @Override
    public String toString() {
        return name();
    }
}
